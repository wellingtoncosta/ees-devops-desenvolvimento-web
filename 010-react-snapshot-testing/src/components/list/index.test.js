import React from 'react'
import renderer from 'react-test-renderer'

import {UsersList} from './'

describe("UsersList tests", () => {
    it("renders correctly when it is loading", () => {
        const domTree = renderer.create(
            <UsersList
                loading={true}
                users={[]}
                error={null}
            />
        ).toJSON();

        expect(domTree).toMatchSnapshot();
    });

    it("renders correctly when there are no users", () => {
        const users = []

        const domTree = renderer.create(
            <UsersList
                loading={false}
                users={users}
                error={null}
            />
        ).toJSON();

        expect(domTree).toMatchSnapshot();
    });

    it("renders correctly when there is an user", async () => {
        const users = [
            {
                id: 1,
                avatar_url: 'https://avataaars.io/',
                login: 'user1',
                name: 'User 1',
            }
        ]

        const domTree = renderer.create(
            <UsersList
                loading={false}
                users={users}
                error={null}
            />
        ).toJSON();

        expect(domTree).toMatchSnapshot();
    });

    it("renders correctly when there is an error", async () => {
        const error = {
            message: 'Unable to load users'
        }

        const domTree = renderer.create(
            <UsersList
                loading={false}
                users={[]}
                error={error}
            />
        ).toJSON();


        expect(domTree).toMatchSnapshot();
    });
});
