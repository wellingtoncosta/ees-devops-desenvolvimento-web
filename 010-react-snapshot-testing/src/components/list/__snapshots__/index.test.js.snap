// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UsersList tests renders correctly when it is loading 1`] = `
<div>
  <nav
    className="bg-dark navbar navbar-expand navbar-light"
  >
    <span
      className="navbar-brand"
    >
      <div
        className="ms-4 text-white fs-3"
      >
        Github Users
      </div>
    </span>
  </nav>
  <div
    className="text-center mt-5"
  >
    <div
      className="spinner-border"
    />
  </div>
</div>
`;

exports[`UsersList tests renders correctly when there are no users 1`] = `
<div>
  <nav
    className="bg-dark navbar navbar-expand navbar-light"
  >
    <span
      className="navbar-brand"
    >
      <div
        className="ms-4 text-white fs-3"
      >
        Github Users
      </div>
    </span>
  </nav>
  <div
    className="text-center mt-5 fs-3"
  >
    There are no results to show.
  </div>
</div>
`;

exports[`UsersList tests renders correctly when there is an error 1`] = `
<div>
  <nav
    className="bg-dark navbar navbar-expand navbar-light"
  >
    <span
      className="navbar-brand"
    >
      <div
        className="ms-4 text-white fs-3"
      >
        Github Users
      </div>
    </span>
  </nav>
  <div
    className="text-center text-danger mt-5 fs-3"
  >
    Unable to load users
  </div>
</div>
`;

exports[`UsersList tests renders correctly when there is an user 1`] = `
<div>
  <nav
    className="bg-dark navbar navbar-expand navbar-light"
  >
    <span
      className="navbar-brand"
    >
      <div
        className="ms-4 text-white fs-3"
      >
        Github Users
      </div>
    </span>
  </nav>
  <div
    className="list-group"
    onKeyDown={[Function]}
  >
    <button
      className="list-group-item list-group-item-action"
      data-rr-ui-event-key={null}
      onClick={[Function]}
    >
      <div
        className="hstack"
      >
        <div
          className="p-2"
        >
          <img
            alt="Avatar"
            style={
              Object {
                "borderRadius": "50%",
                "width": "4em",
              }
            }
          />
        </div>
        <div
          className="p-2"
        >
          <div
            className="fw-semibold"
          >
            User 1
          </div>
          <div
            className="fw-light"
          >
            @
          </div>
        </div>
      </div>
    </button>
  </div>
</div>
`;
